chef/cookbooks/monasca/recipes/master.rb

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2017 Fujitsu LIMITED
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitation.
+
+package "ansible"
+
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+directory "/opt/fujitsu" do
+  owner "root"
+  group "root"
+  mode "0755"
+end
+
+# TODO: use package "monasca-installer"
+remote_file "/opt/fujitsu/cmm-suse.tgz" do
+  source node[:monasca][:master][:cmm_tarball_url]
+  owner "root"
+  group "root"
+  mode "0644"
+  action :create_if_missing
+end
+
+execute "extract cmm tarball" do
+  command "tar xf cmm-suse.tgz"
+  cwd "/opt/fujitsu/"
+  not_if { Dir.exist?("/opt/fujitsu/monasca-installer") }
+  notifies :run, "execute[run ansible]", :delayed
+end
+
+template "/opt/fujitsu/monasca-installer/credentials.yml" do
+  source "credentials.yml.erb"
+  owner "root"
+  group "root"
+  mode "0644"
+  variables(
+    keystone_settings: keystone_settings
+  )
+  notifies :run, "execute[run ansible]", :delayed
+end
+
+template "/opt/fujitsu/monasca-installer/cmm-hosts" do
```

Comment:
General question: do we really need to run the installer through ssh or could we also run it directly on the target machines? 

Because if running it directly on the target machines is possible we should at least consider doing that. This would allow us to easily mix Crowbar resources on the target machine with one or more exec resources for ansible runs and declare dependencies between them. It would also make a later transition to full Crowbar management easier because then we could simply move things from the Ansible installer to its encapsulating Chef recipe.

Please tell me right away if that's a stupid idea that will cause a lot of extra work :-)
Also, I just want to discuss this for now, this is explicitly _not_ a request to modify the code in that direction. If and when we decide it's feasible and a good idea we can do this in a follow-up.